core/src/main/scala/org/apache/spark/rdd/SequenceFileRDDFunctions.scala

```diff
@@ -32,16 +32,13 @@ import org.apache.spark.internal.Logging
  * @note This can't be part of PairRDDFunctions because we need more implicit parameters to
  * convert our keys and values to Writable.
  */
-class SequenceFileRDDFunctions[K <% Writable: ClassTag, V <% Writable : ClassTag](
+class SequenceFileRDDFunctions[K: IsWritable: ClassTag, V: IsWritable: ClassTag](
```

Comment:
Out of curiosity why does this require a new type def?

Reply:
Maybe it's not necessary. I found this is a way won't cause the MIMA tests check fail,  I'll try another way :)

Reply:
I tried three ways to fix issue:

The 1st way:

```
class SequenceFileRDDFunctions[K: ClassTag, V: ClassTag](
    self: RDD[(K, V)],
    _keyWritableClass: Class[_ <: Writable],
    _valueWritableClass: Class[_ <: Writable])
    (implicit kToWritable: K => Writable, vToWritable: V => Writable)
```

The 2nd way:

```
class SequenceFileRDDFunctions[K, V](
    self: RDD[(K, V)],
    _keyWritableClass: Class[_ <: Writable],
    _valueWritableClass: Class[_ <: Writable])
    (implicit ev$1: K => Writable, ev$2: V => Writable, ev$3: ClassTag[K], ev$4: ClassTag[V])
```

The 2 above way will let `MIMIA tests` check failed , and the  3rd way is refer to Scala user [QA](https://users.scala-lang.org/t/scala-2-13-0-deprecates-but-how-do-i-get-rid-of-this-in-a-class-definition/4679),  def a new type `IsWritable` and use `K: IsWritable: ClassTag` instead of  `K <% Writable` and this way not break the `MIMIA tests`.

But I'm very sorry, I haven't figured out why the 3rd way didn't break the `MIMA test` check

Reply:
And I think we could combine the type declaration with the `SequenceFileRDDFunctions ` definition,  but it seems that Spark tells me `avoid using structural types.`

Reply:
I see, yeah we can't add implicits without changing the signature. Hm, I can't think of other ways to write it. I think this change is OK. I'd leave a note in the comments of IsWritable about why it exists, with a link to the suggestion from scala Q&A.